lib/asciidoctor-bibtex/js/bibtex.rb

```diff
@@ -0,0 +1,35 @@
+module AsciidoctorBibtex
+  `const {Cite, plugins} = require('@citation-js/core')`
+  `require('@citation-js/plugin-bibtex')`
+  
+  module BibTeX
+    def self.open(path, options = {})
+      file = File.read(path, encoding: 'utf-8')
+      return Bibliography.new(`new Cite(#{file})`)
+    end
+
+    class Bibliography
+      def initialize(js_bibliography)
+          @js_bibliography = js_bibliography
+      end
+
+      def to_citeproc(options = {})
+        return @js_bibliography
+      end
+
+      def [](key)
+        return Entry.new(%x{#{@js_bibliography}.format('bibtex', { format: 'object'}).find(cite => cite.label === #{key})})
```

Comment:
Would it be more efficient to create a list of entries (only once) in the constructor?

Reply:
Yes, very good idea! I've implemented something in my latest commit, can you have a look?